lib/card/set_pattern.rb

```diff
@@ -91,12 +97,17 @@ def module_key
       end
     end
 
+    def inherited_key modules_hash
+      module_key && modules_hash[ module_key ] ||
+        @inherited_type_module_key && modules_hash[ @inherited_type_module_key ]
```

Comment:
We could avoid setting this when none of the hashes have an entry for this key, but this test isn't all that expensive anyway.